pkgs/top-level/stage.nix

```diff
@@ -8,7 +8,15 @@
    arguments. Normal users should not import this directly but instead
    import `pkgs/default.nix` or `default.nix`. */
 
-
+let
+  # Cache mapping from attributes to unit directories up here so it can get
+  # reused every time this file is called
+  unitDirs = builtins.foldl' (result: shard:
+    result // builtins.mapAttrs (name: type:
+      ../unit + "/${shard}/${name}/pkg-fun.nix"
```

Comment:
```suggestion
      ../unit + "/${shard}/${name}"
```
Why not keep default.nix as a filename?

Reply:
See https://github.com/nixpkgs-architecture/simple-package-paths#alternate-pkg-funnix-filename, feel free to open issues in that repository for feedback :)

Reply:
https://github.com/nixpkgs-architecture/simple-package-paths/issues/30